core/templates/dev/head/services/PromoBarService.js

```diff
@@ -0,0 +1,34 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for Promo Bar.
+ */
+
+oppia.factory('PromoBarService', ['$http',
+  function($http) {
+    return {
+      getPromoBar: function() {
+        return $http.get('/promo_bar_handler', {}).then(
```

Comment:
Check for ENABLE_PROMO_BAR and add ENABLE_PROMO_BAR to dependencies here since if the promobar is disabled in constants we don't need to make the AJAX request.

Reply:
Done. I added the ENABLE_PROMO_BAR dependency in PromoBarDirective.js